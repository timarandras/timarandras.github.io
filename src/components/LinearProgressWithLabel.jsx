import { Box, LinearProgress, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export default function LinearProgressWithLabel({ value, ...props }) {
  return (
    <Box display="flex" alignItems="center">
      <Box width="100%" mr={1}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <LinearProgress variant="determinate" value={10 * value} {...props} />
      </Box>
      <Box minWidth={35}>
        <Typography variant="body2" color="textSecondary">
          {value}
          /10
        </Typography>
      </Box>
    </Box>
  );
}

LinearProgressWithLabel.propTypes = {
  value: PropTypes.number.isRequired,
};
