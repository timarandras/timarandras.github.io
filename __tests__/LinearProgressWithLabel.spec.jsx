import { render } from '@testing-library/react';
import React from 'react';

import LinearProgressWithLabel from '../src/components/LinearProgressWithLabel';

describe('LinearProgressWithLabel', () => {
  test('Matches snapshot', () => {
    const { container } = render(<LinearProgressWithLabel value={5} />);

    expect(container.firstChild).toMatchSnapshot();
  });
  test('Renders a 50% progress bar', () => {
    const { container } = render(<LinearProgressWithLabel value={3} />);
    expect(container.querySelector('.MuiLinearProgress-bar')).toHaveAttribute('style', 'transform: translateX(-70%);');
  });
});
