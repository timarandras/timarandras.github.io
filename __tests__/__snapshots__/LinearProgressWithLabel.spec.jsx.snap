// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearProgressWithLabel Matches snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style="transform: translateX(-50%);"
      />
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      5
      /10
    </p>
  </div>
</div>
`;
